### PR TITLE
Add package i2c-tools

### PR DESCRIPTION
**Pull request by eirikb**
Specify library name and version:  **i2c-tools/4.3**

Hi. My end goal is to push [ddcutil](https://github.com/rockowitz/ddcutil).
To do this I need at least two new packages: **i2c-tools** and **libkmod** (both in kernel.org).

I will see if I can get this package accepted before I try to push libkmod, and eventually ddcutil.  
This isn't my first PR, but it is my first PR with a new package.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

No hooks tested, but I have tried multiple different builds: https://github.com/eirikb/proof-of-conan/actions/runs/1476530986

### CODE REVIEW

File: recipes/i2c-tools/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import os
+from conans import ConanFile, CMake, tools, AutoToolsBuildEnvironment
+
+
+class I2cConan(ConanFile):
```
Comment (madebr) [inline recipes/i2c-tools/all/conanfile.py:5]:
```suggestion
from conans import ConanFile, CMake, tools, AutoToolsBuildEnvironment

requires_conan_version = ">=1.33.0"


class I2cConan(ConanFile):
```

File: recipes/i2c-tools/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    name = "i2c-tools"
+    license = "LGPL 2.1"
+    author = "Eirik Brandtzæg eirikb@eirikb.no"
```
Comment (madebr) [inline recipes/i2c-tools/all/conanfile.py:8]: cci does not add authors to the recipes.
We assume cci is the author.
```suggestion
```

File: recipes/i2c-tools/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    url = "https://mirrors.edge.kernel.org/pub/software/utils/i2c-tools/"
+    description = "I2C tools for the linux kernel as well as an I2C library."
+    topics = ("i2c")
+    os = "linux"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
```
Comment (madebr) [inline recipes/i2c-tools/all/conanfile.py:15]:
```suggestion
    options = {
        "shared": [True, False],
        "fPIC": [True, False],
    }
    default_options = {
        "shared": False,
        "fPIC": True,
    }
```

File: recipes/i2c-tools/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    def _source_subfolder(self):
+        return "i2c-tools"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "i2c-tools-{}".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        self.run("make", cwd=self._source_subfolder)
```
Comment (madebr) [inline recipes/i2c-tools/all/conanfile.py:32]: Please use the autotools build helper. It allows overriding the make program through `CONAN_MAKE_PROGRAM`.
```suggestion
        autotools = AutotoolsBuildEnvironment(self)
        autotools.make(target="libi2c.so" if self.options.shared else "libi2c.a")
        
```

File: recipes/i2c-tools/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    def package(self):
+        self.copy("*.h", dst="include", src=self._source_subfolder, keep_path=False)
+        self.copy("*.h", dst="include/i2c", src=self._source_subfolder, keep_path=False)
+        self.copy("*.so", dst="lib", keep_path=False)
+        self.copy("*.so.0", dst="lib", keep_path=False)
+        self.copy("*.dylib", dst="lib", keep_path=False)
```
Comment (madebr) [inline recipes/i2c-tools/all/conanfile.py:39]: Linux does not have dylib.
```suggestion
```

File: recipes/i2c-tools/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+conan_basic_setup()
+
+add_executable(example example.cpp)
+target_link_libraries(example ${CONAN_LIBS})
+
+# CTest is a testing tool that can be used to test your project.
+# enable_testing()
+# add_test(NAME example
+#          WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/bin
+#          COMMAND example)
```
Comment (madebr) [inline recipes/i2c-tools/all/test_package/CMakeLists.txt:14]:
```suggestion
include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

find_package(i2c-tools REQUIRED CONFIG)

add_executable(example example.cpp)
target_link_libraries(example PRIVATE i2c-tools::i2c-tools)
```

File: recipes/i2c-tools/all/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+    def imports(self):
+        self.copy("*.dll", dst="bin", src="bin")
+        self.copy("*.dylib*", dst="bin", src="lib")
+        self.copy('*.so*', dst='bin', src='lib')
+
+    def test(self):
+        if not tools.cross_building(self):
+            os.chdir("bin")
+            self.run(".%sexample" % os.sep)
```
Comment (madebr) [inline recipes/i2c-tools/all/test_package/conanfile.py:25]:
```suggestion
from conans import ConanFile, CMake, tools
import os


class TestPackageConan(ConanFile):
    settings = "os", "arch", "compiler", "build_type"
    generators = "cmake", "cmake_find_package_multi"

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def test(self):
        if not tools.cross_building(self):
            bin_path = os.path.join("bin", "example")
            self.run(bin_path, run_environment=True)
```

File: recipes/i2c-tools/all/test_package/example.cpp [hunk 8]
```diff
@@ -0,0 +1,4 @@
+#include "i2c/smbus.h"
+
+int main() {
```
Comment (madebr) [inline recipes/i2c-tools/all/test_package/example.cpp:3]: Please exercise the library: use some symbols.

File: recipes/i2c-tools/all/conanfile.py (hunk 2 above)
Comment (eirikb) [reply recipes/i2c-tools/all/conanfile.py:10]: Thanks. I was really wondering about this one.
I'm not the author of the library either.
Just used the defaults from conan new

**Comment by madebr** [comment]
Please consider https://github.com/eirikb/conan-center-index/pull/1

File: recipes/i2c-tools/all/test_package/example.cpp (hunk 8 above)
Comment (eirikb) [reply recipes/i2c-tools/all/test_package/example.cpp]: I'm not (at all) very familiar with the library, as I said in the PR I only need it in order to get another library.
I found some examples online, but do you perhaps think the code you provided in your PR to my branch is sufficient?
Oh and thanks a bunch by the way! :)

Comment (madebr) [reply recipes/i2c-tools/all/test_package/example.cpp]: A test package is just a proof that the library can be used.
I copied the way of initialization from a source inside the tar ball.

No problem.

Comment (eirikb) [reply recipes/i2c-tools/all/test_package/example.cpp]: I guess it's fine then? All I have to do now is to wait for my user to get access to CI to run and build?

Comment (madebr) [reply recipes/i2c-tools/all/test_package/example.cpp]: Yup. They do that in batches every week.
So if in 1 week you've not been accepted yet, post something here.

Comment (eirikb) [reply recipes/i2c-tools/all/test_package/example.cpp]: Great, then I will wait :)

**Comment by eirikb** [comment]
@madebr Hi I think I have been given access now, how do I trigger the build?

**Comment by madebr** [comment]
By pushing some changes or closing/re-opening this pr.

**Comment by eirikb** [comment]
> By pushing some changes or closing/re-opening this pr.

Ah. Would close/re-open mess something up, or is that safe to do?

**Comment by SSE4** [comment]
> Ah. Would close/re-open mess something up, or is that safe to do?

it's safe. I've triggered a build for you. we have a bot, which should restart builds automatically, but it's currently broken, the next deployment should have a fix. sorry for the inconvinience.

**Comment by madebr** [comment]
> > By pushing some changes or closing/re-opening this pr.
> 
> Ah. Would close/re-open mess something up, or is that safe to do?

It only causes mail spam :smile:

**Review by prince-chrismc — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Pull request merged**